### ISSUE TITLE
Tree-Sitter and lsp not working 2.0

### ISSUE THREAD

**Issue by A-Lamia** [AstroNvim/AstroNvim#865]
**Checklist**
Please confirm you have run the following commands and restarted Neovim and are encountering the issue with the current latest version of AstroNvim
- [x] `:PackerSync`
- [x] `:AstroUpdate`
- [x] restarted AstroNvim
- [x] Neovim version >= 0.7.0

**OS**
Windows 10

**Describe the bug**
Tree-Sitter and lsp not working on complete fresh nvim 8.0 and astronvim, neovim syntax highlight works with python and some other files it supports so its not that.

**To Reproduce**
1. uninstalled and deleted everything for nvim and astronvim
2. install 8.0, git cloned astronvim 
3. started with `+PackerSync` then restarted
4. installed TS lua and lua lsp through mason then restarted
5. open lua file TS not working and LSP not connected
6. LSPInfo lua is there but no client connection
7. try running `TSEnable lua` nothing happens
8. try running `LspStart` nothing happens.

**Expected behavior**
LSP and Tree-Sitter should be running.

**Screenshots**
![image](https://user-images.githubusercontent.com/6450181/183763423-c73688e9-beeb-4342-a1a6-7532b124d185.png)
![image](https://user-images.githubusercontent.com/6450181/183763513-cf11fa9e-5284-4be7-968a-d35991bbc571.png)
![image](https://user-images.githubusercontent.com/6450181/183763744-e8a3032a-219e-4a9a-af0a-6356b0137744.png)

**Additional context**

**Comment by mehalter** [comment]
What version of AstroNvim are you using? Neovim 0.8 is an unsupported version. We do have the I don't remember`v2` branch that is as close as you can get to - something working, but it might not be up to date. The current stable and nightly AstroNvim does not supported Neovim 0.8 treesitter or LSP

**Comment by A-Lamia** [comment]
I was using astrovim v2.

**Comment by A-Lamia** [comment]
v2 branch is buggy but everything is working.

**Comment by mehalter** [comment]
What commit of Neovim 0.8 are you using?

**Comment by A-Lamia** [comment]
@mehalter latest im back on 7.2 and AstroNvim nightly was havining issues with v2 and hairline or what ever that plugin is called lol.

**Comment by mehalter** [comment]
Yeah there are a lot of big bugs happening right now in Neovim 0.8 and it's super super broken. Especially winbar: https://github.com/neovim/neovim/issues/19458

I would definitely not recommend using Neovim nightly as it breaks a lot and that's expected currently with `v2` especially since we aren't actively testing it or fixing any bugs yet.

**Comment by A-Lamia** [comment]
@mehalter yeah i just needed it to test out some issues i was having, do you also have any good method for passing params to a local function with a map? eg.
```
["B"] = {thisFunc("arg1", "arg2")}
```

**Comment by mehalter** [comment]
Is this for setting a key binding where you want to use the parameters in the function you call when you press `B` for example?

**Comment by A-Lamia** [comment]
yeah

**Comment by mehalter** [comment]
You can just wrap it in an anonymous function

```lua
B = { function() thisFunc("arg1", "arg2") end },
```

**Comment by A-Lamia** [comment]
aaah interesting, I'll give it a try thanks!

**Comment by A-Lamia** [comment]
@mehalter awesome, you legend works great! thanks :)
